activesupport/lib/active_support/cache.rb

```diff
@@ -844,14 +846,16 @@ module Loader
         extend self
 
         def load(payload)
+          raise InvalidCacheItemError, %{Payload is not a string, is a #{payload.class.name}} unless payload.respond_to?(:start_with?)
```

Comment:
I think you could just cast to entry directly here. Might as well avoid the need for a new error.

Reply:
I was trying to avoid interfering with cache stores other than Memcached, since (I think) this is specifically an issue if you go Dalli -> Memcached. Do you think I should just not worry about that?

Reply:
Also I still need the error handling regardless, to handle the "Invalid cache prefix" item error.

Reply:
> I was trying to avoid interfering with cache stores other than Memcached

Good point. Then you can override `load` in `MemcachedStore` that will make it even more obvious that it's a MemCached specific concern.

Reply:
Okay, I refactored it to avoid raising an error.